src/ipc/ipc4/helper.c

```diff
@@ -472,8 +502,6 @@ int ipc_comp_connect(struct ipc *ipc, ipc_pipe_comp_connect *_connect)
 	 * in case of DP -> LL
 	 *	size = 2*ibs of destination (LL) module. DP queue will handle obs of DP module
 	 */
-	uint32_t buf_size;
-
 	if (source->ipc_config.proc_domain == COMP_PROCESSING_DOMAIN_LL)
 		buf_size = source_src_cfg.obs * 2;
```

Comment:
shouldn't this be changed to obs * 2? and similarly ibs * 2 below?

Reply:
but more simply isnt it enough to do MAX(ibs * 2, obs * 2)?

Reply:
@andyross not changing the buffer sizing according to the ibs/obs based on pin numbers would be incorrect isnt it? Let's assume we're binding modules with a non-zero pin ID's, what would sink_src_cfg.ibs or source_src_cfg.obs contain?

Reply:
Oh.... gotcha.  I had missed that the calculation here was still done with the source_src_cfg value and not the recalculated-and-now-pin-aware values from above.   Interestingly, this patch actually started as a refactor of the code in mtl-007-drop-stable, and it actually has exactly the same bug: https://github.com/thesofproject/sof/blob/mtl-007-drop-stable/src/ipc/ipc4/helper.c#L494

Reply:
I dont think mtl-007 has the bug. in that patch, the buf_size is calculated from the ibs/obs set based on the pin ID i.e from the base config for pin ID 0 or from the base config extn if pin ID > 0

Reply:
Should be fixed.  Please check.